### DISCUSSION TITLE
Using eclipse preferences files and formatter configuration

### DISCUSSION THREAD

**Discussion by gnfisher** [mfussenegger/nvim-jdtls#236] [Q&A] [answered]
Hello, my employer provides an eclipse *.epf file as well as a *.xml file for formatter configurations. I am unfamiliar with eclipse and new to Java and had little luck poking around docs to see if I could use these somehow with the jdtls lsp and nvim-jdtls. I was hoping you might know if it were possible and if so, if you could point me in the direction I ought to go. Thanks!

**Comment by mfussenegger** [comment, marked as answer]
Is https://github.com/mfussenegger/nvim-jdtls/issues/203#issuecomment-1006634815 what you're looking for?

**Comment by gnfisher** [comment]
Yes, this answers my formatting question. :facepalm: Apologies I didn't dig that up on my own.

--
  Greg Fisher
  ***@***.***

**Comment by gnfisher** [comment]
Okay, so I've got this all hooked up as described. Results of `vim.lsp.buf.formatting()` change when I include:

```
          format = {
            settings = { url = home .. '/.local/share/eclipse_format_pref.xml', -- expects xml settings placed here
            profile = 'myprofile' }
          },
```
However, formatting is changing 2 spaces to 4 spaces, which ignores every setting I've placed!

I've set shiftwidth/tabstop/sottabstop to 2 in ftplugins, I've tried passing in tabSize to formatting like `vim.lsp.buf.formatting({ tabSize = 2 })`. The settings xml also specifies 2 spaces indentation. 

I've checked logs and I see this:

```
[DEBUG][2022-01-30 18:39:35] .../lua/vim/lsp.lua:962    "LSP[jdt.ls]"   "client.request"        1       "textDocument/formatting"     {  options = {    insertSpaces = true,    tabSize = 2  },  textDocument = {    uri = "file:///..myfile.java"  }}        <function 1> 
```
So it seems at least some "tabsize" setting is getting piped in?

Any ideas why its still giving me 4 spaces indentation? I just can't sort it out.

**Reply by gnfisher** [reply]
I've side stepped this by using java prettier for now.

**Reply by mfussenegger** [reply]
For me your example with calling `vim.buf.formatting` works as expected:


https://user-images.githubusercontent.com/38700/152504328-074cbafd-7d53-4348-ac64-3b464f1266ff.mp4

**Reply by gnfisher** [reply]
Surely something in my config! I've set up java prettier in any case now so its not an issue. Thank you so much for all your work and support. I'm using vim again after a few weeks in IDE land and it's so nice to be back.

**Comment by ervinpm** [comment]
Do you know if there is a way to specify eclipse jdt version for the formatter? For some reason the formatting differs for each eclipse version even when using the same xml file.
